### ISSUE TITLE
Spot on Azure

### ISSUE THREAD

**Issue by yellowhat** [hashicorp/packer#8776]
Hi,
is it possible to use packer on Azure using spot instances?

#### Feature Description
Use spot instances on Azure

#### Use Case(s)
For devops purpose, spot instance can be a good way to reduce costs.

Thanks

**Comment by jonmaestas** [comment]
This is available in the AWS `amazon-ebs` builder. Makes sense to add it to the Azure builders as well.
```
...
"spot_price": "auto",
"spot_price_auto_product": "Linux/UNIX (Amazon VPC)",
...
```
https://www.packer.io/docs/builders/amazon-ebs.html#spot_price
https://www.packer.io/docs/builders/amazon-ebs.html#spot_price_auto_product

**Comment by SwampDragons** [comment]
I agree this would be nice to add; I'll try to find room for it in the next couple of releases, but I'm not sure how technically involved it will be so it may take a while.
